docs/HTTP3.md

```diff
@@ -210,7 +214,7 @@ Build curl:
 
 QUIC support is **EXPERIMENTAL**
 
-Build OpenSSL 3.3.1:
+Build OpenSSL 3.3.1 or newer:
```

Comment:
CMake says 3.3.0: 8d067e61b0a1ac2969cd2f119b134c2743f1b0af. Should we sync that?

Reply:
3.3.0 *may* work, but I think there were some important things fixed in 3.3.1, like UDP sockets for macOS as one thing I do remember.